### PR TITLE
HMA 4886

### PR DESCRIPTION
**Pull request by chrisob55**
https://jira.tools.tax.service.gov.uk/browse/HMA-4886

### CODE REVIEW

**Review by mrolfe88 — CHANGES_REQUESTED** [review]
Just a couple of minor changes, you will need to edit the definition.scala file though as they are managing scopes differently on the platform now and a scope needs to have the same name and description in ever service it is used. See [here](https://github.com/hmrc/mobile-self-assessment/blob/main/app/uk/gov/hmrc/mobileselfassessment/views/definition.scala.txt) for reference. The API will fail to publish currently.

Other less important things that will need doing at some point:

- Add example response json to Raml
- Add Shuttering support

File: app/uk/gov/hmrc/mobilepayments/connectors/ShutteringConnector.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  @Named("mobile-shuttering") serviceUrl: String) {
+
+  val logger: Logger = Logger(this.getClass)
+
+  def getShutteringStatus(
+    journeyId:              JourneyId
+  )(implicit headerCarrier: HeaderCarrier,
+    ex:                     ExecutionContext
+  ): Future[Shuttering] =
+    http.GET[JsValue](s"$serviceUrl/mobile-shuttering/service/mobile-self-assessment/shuttered-status?journeyId=$journeyId").map {
```
Comment (mrolfe88) [inline app/uk/gov/hmrc/mobilepayments/connectors/ShutteringConnector.scala:42]: copy/paste typo here "mobile-self-assessment"

I'd suggest going with mobile-open-banking as the service name to give us a bit of flexibility in how we shutter the different parts of this service.

Comment (chrisob55) [reply app/uk/gov/hmrc/mobilepayments/connectors/ShutteringConnector.scala:42]: Ah yes - copy and paste error there. Cheers

File: it/stubs/ShutteringStub.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      ).willReturn(
+          aResponse()
+            .withStatus(200)
+            .withBody(s"""
+                         |{
+                         |  "shuttered": true,
+                         |  "title":     "Shuttered",
+                         |  "message":   "Self-Assessment is currently not available"
+                         |}
+          """.stripMargin)
```
Comment (mrolfe88) [inline it/stubs/ShutteringStub.scala:42]: Same as above

**Review by mrolfe88 — APPROVED** [review]

**Pull request merged**